core/trino-main/src/main/java/io/trino/connector/system/TrinoMaterializedViewAdapter.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.connector.system;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.SchemaTableName;
+
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
+
+public class TrinoMaterializedViewAdapter
+        implements MaterializedViewRowAdapter
+{
+    @Override
+    public ConnectorTableMetadata getTableLayout()
+    {
+        return new ConnectorTableMetadata(getTableName(), getTableColumns());
+    }
+
+    @Override
+    public SchemaTableName getTableName()
+    {
+        return new SchemaTableName("metadata", "materialized_views");
+    }
+
+    @Override
+    public List<ColumnMetadata> getTableColumns()
+    {
+        return ImmutableList.of(
+                new ColumnMetadata("catalog_name", createUnboundedVarcharType()),
+                new ColumnMetadata("schema_name", createUnboundedVarcharType()),
+                new ColumnMetadata("materialized_view_name", createUnboundedVarcharType()),
```

Comment:
let's call this `name` since we're not using the `materialized_view_` prefix in the other column names